pkg/pipelines/secrets/secrets.go

```diff
@@ -159,6 +164,22 @@ func createDockerConfigSecret(name types.NamespacedName, in io.Reader) (*corev1.
 	return createSecret(name, ".dockerconfigjson", corev1.SecretTypeDockerConfigJson, in)
 }
 
+func createBasicAuthSecret(name types.NamespacedName, token, host string) *corev1.Secret {
+	return &corev1.Secret{
+		TypeMeta: secretTypeMeta,
+		ObjectMeta: meta.ObjectMeta(name, meta.AddAnnotations(
+			map[string]string{
+				"tekton.dev/git-0": host,
+			}),
+		),
+		Type: corev1.SecretTypeBasicAuth,
+		StringData: map[string]string{
+			"username": "tekton",
+			"password": token,
+		},
+	}
+}
+
```

Comment:
CreateSealedBasicAuthSecret() and createBasicAuthSecret() should probalby be named to indicate they are not  "generic" functionas to create basic auth secret but they are specically to BA secret for tekton.

Reply:
I can rework to make this a bit more generic.